mitxgraders-js/MJxPrep.test.js

```diff
@@ -103,12 +105,24 @@ describe('preProcessEqn', () => {
 
   it('replaces cross products', () => {
     const eqn = 'cross(x) + cross(a + b, c) + y'
-    expect(preProcessEqn(eqn)).toBe('cross({:x:})+{:({:a:}+{:b:}) {:times:} {:c:}:}+{:y:}')
+    expect(preProcessEqn(eqn)).toBe('{:cross({:x:}):}+{:({:a:}+{:b:}) {:times:} {:c:}:}+{:y:}')
   } )
 
+  it('wraps variables and function calls', () => {
+    // The behavior of wrapVariables and wrapFunCalls is primarily described
+    // in their dedicated tests
+    const expr = "x + f'(x, y) + cross(a, g(b)+2)/h''(y) + sin(pi/4)^3 + 5"
+    const expected = "{:x:}+{:f'({:x:},{:y:}):}+{:{:a:} {:times:} ({:g({:b:}):}+2):}/{:h''({:y:}):}+{:sin({:pi:}/4):}^3+5"
+    expect(preProcessEqn(expr)).toBe(expected)
+  } )
+
+} )
+
+describe('wrapVariables', () => {
+
   it('wraps variable names in unbreakable invisible brackets', () => {
     const exprs = [
-      ["f''/f + g_a'+sin(horse)", "{:f'':}/{:f:}+{:g_a':}+sin({:horse:})"],
+      ["f''/f + g_a' + sin(horse)", "{:f'':}/{:f:} + {:g_a':} + sin({:horse:})"],
```

Comment:
Wait, how does this work? I thought I stripped spaces out of the entry so that it wouldn't mess with the regex?

Reply:
Note that I moved your variable wrapping into `wrapVariables`, and this only tests the `wrapVariables` function. So `preprocessEqn` still strips whitespace before wrapping the variables.

That said, what part of `preprocessEqn` requires whistepsace-stripping? Do we really need to strip whitespace? I am not sure...

Reply:
How do you think `f (x)` should render? What about `f_{mu nu}?` There are just a bunch of places where a space might be inserted, and I didn't want to deal with this in the regex. Besides, the first thing we do in the processing is to strip spaces... Removing them just made my thought processes much more concrete.

Reply:
I think you're right. Stripping whitespace will make `f (x)/2` render correctly, so let's stick with it.